### PR TITLE
Update Node to v20.9.0

### PR DESCRIPTION
**Pull request by DavidBiddle**
### What problem does this pull request solve?

Trello card: https://trello.com/c/EIvl0xBy/1264-update-node-to-v209

Currently blocked by the Alpine image not having 20.9 available yet.

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ WORKDIR /app
 
 RUN apk update
 RUN apk upgrade --available
-RUN apk add libc6-compat openssl-dev build-base libpq-dev nodejs=~18 npm git python3
+RUN apk add libc6-compat openssl-dev build-base libpq-dev nodejs=~20 npm git python3
```
Comment (aliuk2012) [inline Dockerfile:7]: I dont think this alpine docker image knows what node 20 is. https://pkgs.alpinelinux.org/packages?name=nodejs&branch=edge&repo=&arch=&maintainer=

we will need to bump the image being used. When I checked last Friday asdf nodejs plugin didn't have node 20 added to it either

Comment (DavidBiddle) [reply Dockerfile:7]: Yeah, v20 is too new so I'm going to leave this in draft

Comment (aliuk2012) [reply Dockerfile:7]: asdf nodejs now has v20.9.0

Comment (aliuk2012) [reply Dockerfile:7]: Alpine still using v18

Comment (DavidBiddle) [reply Dockerfile:7]: I've made a backlog card for updating it and added a note about Alpine not updating yet: https://trello.com/c/EIvl0xBy/1264-update-node-to-v209

There's a [Gitlab PR open for the update](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/54105), I assume it'll be at least few days before that gets merged and released.

Comment (aliuk2012) [reply Dockerfile:7]: @DavidBiddle node 20 should be available for alpine now

Comment (DavidBiddle) [reply Dockerfile:7]: Hmm still needs more looking at I think - getting this build failure:

```
> [build  5/16] RUN apk add libc6-compat openssl-dev build-base libpq-dev nodejs=~20 npm git python3:
0.547 ERROR: unable to select packages:
0.548   nodejs-current-20.8.1-r0:
0.548     breaks: world[nodejs~20]
0.548     satisfies: npm-9.6.6-r0[cmd:node]
```

Comment (aliuk2012) [reply Dockerfile:7]: 😭  so at the moment only alpine edge has nodejs 20.9........ alpine 3.18 hasn't been updated yet to support it.

Also  we will need to update the docker image being pulled in from `Dockerfile`

**Comment by DavidBiddle** [comment]
Superseded by #947

**Pull request closed without merging**